pinecone_text/dense/base_dense_ecoder.py

```diff
@@ -1,8 +1,18 @@
-from typing import List, Union
+from typing import List, Union, Any, Optional
 from abc import ABC, abstractmethod
+from functools import cached_property
 
 
 class BaseDenseEncoder(ABC):
+    def __init__(self, *, dimension: Optional[int] = None, **kwargs: Any):
```

Comment:
I think that if we put dimension as optional param for the base class, we should either allow it in all the child classes or not in the base at all. Right now it seems we only going to allow it for OpenAI, so maybe better to not put it in the base class? WDYT?

Reply:
I want to have dimension accessible to all the child classes since it will be weird if only OpenAI exposes. You are right in the fact that in the base init we should not receive this parameter. I create a private property with a default now and override in OpenAI now. I only define as an initialization parameter in OpenAI one.